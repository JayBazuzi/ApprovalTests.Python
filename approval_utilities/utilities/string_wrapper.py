class StringWrapper:
    def __init__(self) -> None:
        self.string = ""

    def append(self, text: str) -> None:
        self.string += text

    def __str__(self) -> str:
        return self.string

    def __repr__(self) -> str:
        return self.string
