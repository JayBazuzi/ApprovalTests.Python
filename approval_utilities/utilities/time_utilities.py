import os

from typing_extensions import ContextManager
from typing import Optional, Type
from types import TracebackType


def use_utc_timezone() -> ContextManager:
    class TimeZoneSwap:
        def __init__(self) -> None:
            self.timezone: Optional[str] = ""

        def __enter__(self) -> None:
            self.timezone = os.environ.get("TZ")
            os.environ["TZ"] = "UTC"

        def __exit__(
            self,
            exc_type: Optional[Type],
            exc_val: Optional[BaseException],
            exc_tb: Optional[TracebackType],
        ) -> bool:
            if self.timezone is None:
                os.environ.pop("TZ")
            else:
                os.environ["TZ"] = self.timezone

    return TimeZoneSwap()
