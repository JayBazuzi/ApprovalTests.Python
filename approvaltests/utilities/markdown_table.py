from typing import Any, Callable, Iterable


from approvaltests.core.verifiable import Verifiable
from approvaltests.core.options import  Options
from approvaltests.core.verify_parameters import VerifyParameters



class MarkdownTable(Verifiable):
    def get_verify_parameters(self, options: Options) -> VerifyParameters:
        return VerifyParameters(options.for_file.with_extension(".md"))

    @staticmethod
    def with_headers(*column_names: str) -> "MarkdownTable":
        table = MarkdownTable()
        table.add_rows(*column_names)
        dividers = map(lambda _: "---", column_names)
        table.markdown += MarkdownTable.print_row(*dividers)
        return table

    def add_rows(self, *column_names):
        self.markdown = MarkdownTable.print_row(*column_names)

    def __str__(self):
        return self.markdown

    @staticmethod
    def print_row(*column_names: Any) -> str:
        row = "|"
        for column in column_names:
            row += f" {column} |"
        return row + "\n"

    def add_rows_for_inputs(self, inputs: Iterable[Any], *varargs: Callable[[Any], Any]) -> "MarkdownTable":
        for input in inputs:
            #Create a list with the input and all the conversions, using varargs.map
            row = [input]
            row += map(lambda callable: callable(input), varargs)
            self.markdown += MarkdownTable.print_row(*row)
        pass