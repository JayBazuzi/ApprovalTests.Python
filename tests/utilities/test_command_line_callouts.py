from pathlib import Path

from approvaltests.utilities.command_line_approvals import (
    verify_command_line_with_inputs,
    verify_command_line,
)


def test_verify_command_line():
    verify_command_line("echo hello world!")


def test_verify_command_line_with_input():
    verify_command_line(
        'python -c "import sys; print(sys.stdin.read())"', input="input"
    )

def test_command_line_verify():
    verify_command_line(
        f'python approvaltests/verify.py -t tests/utilities/hello', input="hello world",
        cwd = Path(__file__).parents[2],
        additional_env= {'PYTHONPATH': '.'}
    )



def test_verify_command_line_with_inputs():
    verify_command_line_with_inputs(
        'python -c "import sys; print(sys.stdin.read())"', inputs=range(3, 7)
    )
