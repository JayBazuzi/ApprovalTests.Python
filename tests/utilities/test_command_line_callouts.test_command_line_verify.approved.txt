Test Passed: tests/utilities/hello
